flecs_ecs_tracing/src/perf_trace.rs

```diff
@@ -0,0 +1,130 @@
+use crate::{metadata, util::*};
+use core::ffi;
+use std::{
+    borrow::Cow,
+    collections::HashMap,
+    sync::{LazyLock, OnceLock, RwLock},
+};
+use tracing_core::{span, Dispatch, Level, Metadata};
+
+/// Log level for performance tracing spans
+pub(crate) static SPAN_LEVEL: OnceLock<Level> = OnceLock::new();
+
+/// Registry of dynamic tracing spans generated from Flecs
+static REGISTRY: LazyLock<RwLock<Registry>> = LazyLock::new(Default::default);
+
+/// Registry of dynamic tracing spans generated from Flecs
+struct Registry {
+    spans: HashMap<
+        String,
+        (tracing_core::span::Id, &'static Metadata<'static>),
```

Comment:
This feels odd. Why's the `Id` separate from the `Metadata`? Or why's this a tuple and not some type?

And why can't someone legitimately own this metadata, like the registry? Is it because it is owned by the dynamic callsite, which you're leaking? If so, why not store that somewhere around here? But you're the one that decided to store it in the callsite, so ... ?

Reply:
`tracing_core` borrows `Callsite` for `'static` and demands that `Callsite` can provide a reference to the `Metadata` (`fn metadata(&self) -> &Metadata<'_>`).

I guess I could store the `MetadataRequest` inside `DynamicCallsite` and do lookups in `REGISTRY` every time but I'm not seeing any advantages to doing that...